docs/components/modeler/element-templates/template-metadata.md

```diff
@@ -0,0 +1,222 @@
+---
+id: template-metadata
+title: Template metadata
+description: "Learn about template metadata fields like name, ID, description, keywords, versioning, and JSON schema compatibility."
+---
+
+The metadata of an element template contains important information about the template itself, such as its name, description, version, compatibility with different Camunda versions, and the schema that is used to validate the template.
+
+## Validation: `$schema`
```

Comment:
@barmac I think the section on the schema feels a little unclear still. Can you think of a precise and useful description that ticks all the boxes? 
I'll add this as a follow up in any case.